### PR TITLE
Replace `lazy_static` with `OnceLock`

### PR DESCRIPTION
**Pull request by stefansundin**
@mediocregopher Feel free to pull this.

I replaced it in the tests and the benchmark even though those files are still out of date and not working. The example seems to still work fine though, which is what matters.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -10,30 +10,55 @@ use hyper::http::uri::InvalidUri;
...
+            trailer_header().clone(),
+            HeaderName::from_static("keep-alive"),
+            HeaderName::from_static("proxy-connection"),
+            HeaderName::from_static("proxy-authenticate"),
+            HeaderName::from_static("proxy-authorization"),
+            HeaderName::from_static("transfer-encoding"),
+            HeaderName::from_static("upgrade"),
+        ]
+    })
 }
```
Comment (stefansundin) [inline src/lib.rs:62]: This got a lot more verbose but I am not sure if there's a better way to do it?

File: src/lib.rs [hunk 2]
```diff
@@ -240,22 +265,22 @@ fn create_proxied_request<B>(
 
         request
             .headers_mut()
-            .insert(&*TE_HEADER, HeaderValue::from_static("trailers"));
+            .insert(te_header(), HeaderValue::from_static("trailers"));
```
Comment (stefansundin) [inline src/lib.rs:268]: Hey. I just realized that maybe this HeaderValue (and the one below for `UPGRADE`) should also use OnceCell?

File: src/lib.rs (hunk 1 above)
Comment (mediocregopher) [reply src/lib.rs:62]: Oof, I guess I didn't realize all of this was wrapped up in `lazy_static` too...

Honestly all this stuff might be better off sitting on the `ReverseProxy` struct. That way there's not some kind of Sync call being incurred on every HTTP request (actually, many Sync calls, one per header which is used!).

But I'm happy to merge this like this for now, it's at least not a regression.

File: src/lib.rs (hunk 2 above)
Comment (mediocregopher) [reply src/lib.rs:268]: Do you mean OnceLock? If so then yeah, probably

**Comment by mediocregopher** [comment]
Just pulled the commit, thanks!

File: src/lib.rs (hunk 1 above)
Comment (stefansundin) [reply src/lib.rs:62]: Hmm, interesting. I don't know a lot about what costs Sync incurs. I guess this is an incentive to get the benchmark file working so we can see if the changes I've made are even observable. I want to learn more about profiling rust code, checking how much memory is allocated for various code paths, etc.

File: src/lib.rs (hunk 2 above)
Comment (stefansundin) [reply src/lib.rs:268]: Yes, you're right, I meant OnceLock. I'll probably make this change later.

**Pull request merged**